file_windows.go

```diff
@@ -320,6 +426,22 @@ func browseForFolderCallback(wnd win.HWND, msg uint32, lparam, data uintptr) uin
 	return 0
 }
 
+func coInitialize() (context.CancelFunc, error) {
+	runtime.LockOSThread()
+	err := win.CoInitializeEx(0, win.COINIT_APARTMENTTHREADED|win.COINIT_DISABLE_OLE1DDE)
+	if err == nil || err == win.S_FALSE {
```

Comment:
If the app calls `CoInitializeEx` with `win.COINIT_APARTMENTTHREADED|win.COINIT_DISABLE_OLE1DDE` before calling `zenity`, the `err` will be `win.S_FALSE`. `zenity` should not call `win.CoUninitialize()`.
So, `err == win.S_FALSE` should be moved to below: `err == win.RPC_E_CHANGED_MODE || err == win.S_FALSE`.

Reply:
I think we've covered this before:

> To close the COM library gracefully on a thread, each successful call to CoInitialize or CoInitializeEx, **_including any call that returns S_FALSE, must be balanced by a corresponding call to CoUninitialize._**

https://learn.microsoft.com/en-us/windows/win32/api/combaseapi/nf-combaseapi-coinitializeex

Reply:
But if you don't want COM to be closed, you can `CoInitializeEx` yourself, and never `CoUninitialize`.
This is a reference count, if you don't `CoUninitialize` it never goes back to zero.